### ISSUE TITLE
Improvement/Bug :: Watermark format incompatibility

### ISSUE THREAD

**Issue by jvahldick** [cshum/imagor#257]
Hi there

I'm facing issues related to the image format when inserting the watermark.
The issue is: "error": "imagor: 406 linear: vector must have 1 or 2 elements"

You can test it with the image attached.
![watermark-image](https://user-images.githubusercontent.com/1113015/203837246-1f92e1f2-e222-4d4a-a1bb-175122c8886f.png)

One idea is to identify if the format is compatible or, in case of failure, try to convert it to the proper format.
I could not figure out what was wrong, but due to this [issue](https://github.com/libvips/libvips/issues/1525), I ended up converting the image to CMYK and the watermark worked.

Would you have any suggestions for this?

**Comment by cshum** [comment]
@jvahldick its because the image is 2 bands i.e. grayscale + alpha. Applied a fix for that

**Comment by cshum** [comment]
You may test the changes at master branch docker:

```
ghcr.io/cshum/imagor@sha256:204c8baaa09c186b54d301b92b7a544073ec982250257ee5b97fd9989be92892
```
https://github.com/cshum/imagor/pkgs/container/imagor/53615615?tag=master

**Comment by cshum** [comment]
Available in v1.3.3

**Comment by jvahldick** [comment]
Thanks @cshum 
In my tests with the provided container image, it was working fine. I would continue some other cases today, but I believe it is good indeed.
